src/skmatter/neighbors/_sparsekde.py

```diff
@@ -0,0 +1,667 @@
+import warnings
+from typing import Callable, Union
+
+import numpy as np
+from numpy.typing import ArrayLike
+from scipy.special import logsumexp as LSE
+from sklearn.base import BaseEstimator
+from sklearn.utils.validation import check_is_fitted, check_random_state
+from tqdm import tqdm
+
+from ..metrics._pairwise import (
+    pairwise_mahalanobis_distances,
+    periodic_pairwise_euclidean_distances,
+)
+from ..utils._sparsekde import effdim, oas
+
+
+class SparseKDE(BaseEstimator):
+    """A sparse implementation of the Kernel Density Estimation.
+    This class is used to build a sparse kernel density estimator.
+    It takes a set of descriptors and a set of weights as input,
+    and fit the KDE model on the sampled point (e.g. the grid point
+    selected by FPS). First, the probability density is estimated for
+    each sampled point. Then, quick shift clustering is applied to the
+    grid points. Finally, a kernel density estimator is built based on
+    the clustering results.
+
+    .. note::
+        Currently only the Gaussian kernel is supported.
```

Comment:
Maybe you still use a `kernel` argument that is `'gaussian'` and if anything else `raise NotImplementedError('SparseKDE only supports Gaussian kernel.')`. Not sure what the plan for the future is.